server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMailboxesMethodTest.java

```diff
@@ -96,6 +97,226 @@ public void teardown() {
     }
 
     @Test
+    public void setMailboxesCreateMailboxShouldSubscribeUser() throws Exception{
+        String requestBody =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"create\": {" +
+                "        \"create-id01\" : {" +
+                "          \"name\" : \"foo\"" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        given()
+            .header("Authorization", this.accessToken.serialize())
+            .body(requestBody)
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxesSet"))
+            .body(ARGUMENTS + ".created", hasKey("create-id01"));
+
+        assertThat(jmapServer.serverProbe().listSubscriptions(username)).containsOnly("foo");
+    }
+
+    @Test
+    public void setMailboxesCreateChildOfInboxMailboxShouldSubscribeUser() throws Exception {
+        String inboxId =
+            with()
+                .header("Authorization", this.accessToken.serialize())
+                .body("[[\"getMailboxes\", {}, \"#0\"]]")
+            .when()
+                .post("/jmap")
+            .then()
+                .extract()
+                .jsonPath()
+                .getString(ARGUMENTS + ".list[0].id");
+
+        String requestBody =
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"create\": {" +
+                "        \"create-id01\" : {" +
+                "          \"name\" : \"foo\"," +
+                "          \"parentId\" : \"" + inboxId + "\"" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        given()
+            .header("Authorization", this.accessToken.serialize())
+            .body(requestBody)
+        .when()
+            .post("/jmap");
+
+        assertThat(jmapServer.serverProbe().listSubscriptions(username)).containsOnly("inbox.foo");
+    }
+
+    @Test
+    public void setMailboxesUpdateMailboxNameShouldChangeSubscriptionUser() throws Exception {
```

Comment:
after some thinking, I think we can consider jmap as being a client that handles its subscriptions, so we can let it like that.